.devcontainer/Dockerfile

```diff
@@ -1,19 +1,23 @@
-FROM python:3.10
+FROM python:3.14
```

Comment:
I appreciate the work to get `python` and `node` to play nicely together.

I'm not totally opposed to moving to `python:3.14` but will just point out we are standardized on `python:3.12` everywhere else.

Maybe this is a good opportunity to tease out any meaningful differences for our dev environment at least.